File: java/json-transform/src/main/java/ai/bizone/jsontransform/functions/TransformerFunctionString.java
package ai.bizone.jsontransform.functions;

import ai.bizone.jsontransform.functions.common.*;
import ai.bizone.jsontransform.functions.common.*;

import java.util.Map;

public class TransformerFunctionString extends TransformerFunction {
    public TransformerFunctionString() {
        super(FunctionDescription.of(
            Map.of(
            "json", ArgumentType.of(ArgType.Boolean).position(0).defaultValue(false)
            )
        ));
    }
    @Override
    public Object apply(FunctionContext context) {
        var value = context.getUnwrapped(null);
        if (context.getBoolean("json")) {
            // although gson.toJson will return "null" eventually, this is quicker
            if (value == null) {
                return "null";
            }
            var adapter = context.getAdapter();
            return adapter.toString(value);
        }
        return context.getAsString(value);
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionString.ts
import TransformerFunction from "./common/TransformerFunction";
import FunctionContext from "./common/FunctionContext";
import { getAsString } from "../JsonHelpers";
import { ArgType } from "./common/ArgType";
import { JSONBig } from "./common/FunctionHelpers";

class TransformerFunctionString extends TransformerFunction {
  constructor() {
    super({
      argsSets: [[{ name: "json", type: ArgType.Boolean, defaultValue: false }]],
    });
  }

  override async apply(context: FunctionContext): Promise<any> {
    const value = await context.getUnwrapped(null);
    if (await context.getBoolean("json")) {
      return JSONBig.stringify(value);
    }
    return getAsString(value);
  }
}

export default TransformerFunctionString;
